doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +119,67 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
+* The defaults are overlayed by the options specified by the selected board in its :file:`*_defconfig` file and its :file:`Kconfig.defconfig` file (see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
+* The configuration is additionally overlayed by the :file:`prj.conf` file provided by the application.
+
+After merging the configuration sets imported from mentioned configuration files, the build system creates a single :file:`.config` file describing entire software configuration of the system.
+This is done once and configuration comes solely from :file:`.config` file until project is reevaluated.
+
+.. note::
+   You can edit :file:`.config` file and rebuild the application to observe changed configuration.
+   This changes will be lost when the application project is re-evaluated.
+   To store the configuration permanently you need to identify which file should provide desired option and edit it directly.
+   In most cases this will be your application :file:`prj.conf` file.
 
 For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
 
+The hardware related configuration comes from ``dts`` files.
+There are several layers providing the hardware definitions:
+
+* The lowest layer results of defaults provided in SoC ``dtsi`` files.
+* The ``dtsi`` files are included by selected board in its ``dts`` file, which provides defaults of hardware configuration from entire board's PoV.
+* The hardware related configuration is further overlayed by the :file:`app.overlay` file provided by the application.
+
+After merging the configuration sets imported from mentioned configuration files, the build system creates a single :file:`zephyr.dts` file describing entire hardware related configuration of the system.
+This process also creates a set of header files files proving defines that can be used to refer hardware configuration from the source code.
+The hardware related configuration does not change until the project is re-evaluated.
+
+.. note::
+   The files resulting from ``dts`` evaluation should not be notified.
+   To store the configuration parmanantly you need to identify file contributing to the specific part of ``dts`` tree and edit it directly.
+   In most cases this will be your application :file:`app.overlay` file.
+
+If enabled the ``partition manager`` provides the memory layout definition.
+The layout is impacted by various elements such as ``Kconfig`` configuration options or presence of child images.
+The build system works to ensure all required partitions are enabled and select right size of each partition area.
+In case you wish to provide an own partition definition, this is done by specifying static configuration in :file:`pm_static.yml` file.
+
+After project evaluation the build system creates a single :file:`partitions.yml` file describing resulting memory layout.
```

Comment:
And `pm.config` which contains constants describing the partition layout, which can be used in the code and build scripts.

Reply:
Yes - defining the rules. I think we spent some time here. I would like @greg-fer to have a look first and see how to split this up. As I mentioned we have some page about building. It simply short and missing multiple things related to what NCS adds. We also need to see if some things that are already said in Zephyr should be stated here (I think to some extend these should be repeated so user has a complete picture of NCS build system).